lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -7089,7 +7090,7 @@ namespace Js
     {
 #if ENABLE_PROFILE_INFO
         // Switch off profiling is asmJsFunction
-        if (this->GetIsAsmJsFunction())
+        if (this->GetIsAsmJsFunction() || this->GetAsmJsModuleInfo())
```

Comment:
because now the pointer is saved in a dynamically expanding structure, so while background thread is accessing, the foreground thread might be expanding the structure, so it can get wrong value. 
so when GetIsAsmJsFunction() returns false, we can still have AsmJsModuleInfo on the function body, right? if so let me submit a PR to use lock for background accessing